### DISCUSSION TITLE
How can I limit the input port connections to 1?

### DISCUSSION THREAD

**Discussion by Biowavix** [clientIO/joint#2098] [Q&A] [answered]
### Introduction

Similar to [this question](https://github.com/clientIO/joint/discussions/2091)
Is there a direct way to limit the Links to the inputs ports to only one? I only reach to a bit tricky solutions that need to iterate over all the graph Links and check if there are two Links pointing to the same target.

Can I do this in a more efficient and direct way? Like checking the port outbound connections or something similar?

Thank you in advance!

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
Please, check out this [example](https://codepen.io/jointjs/pen/rNdzYZj) (`validateConnection()`).

```js
if (
  graph
    .getConnectedLinks(target, { inbound: true })
    .find((link) => link.target().port === targetPort)
) {
  // The port has already an inbound link (we allow 1 link per port inbound port)
  return false;
}
```
